src/auth/auth.controller.ts

```diff
@@ -1,41 +1,46 @@
-import { Body, Controller, Get, Param, Post, Req, Res, UseGuards } from '@nestjs/common';
+import { Body, Controller, Get, Param, Post, Req, Res, UseGuards, Query, Headers } from '@nestjs/common';
 import { RegisterDto } from '../dtos/register.dto';
 import { LoginDto } from '../dtos/login.dto';
 import { AuthService } from './auth.service';
 import { TokenAuthGuard } from './guards/token.auth.guard';
-import { AuthGuard } from '@nestjs/passport';
 import { SessionHandler } from './handlers/session.handler';
+import { OauthHandler } from './handlers/oauth.handelr';
+import { SystemInfo } from 'src/decorators/system-info';
+import { Session } from 'src/entities/session.entity';
 
 @Controller('auth')
 export class AuthController {
   constructor(
     private readonly authService: AuthService,
-    private readonly sessionHandler: SessionHandler
+    private readonly sessionHandler: SessionHandler,
+    private readonly oauthHandler: OauthHandler
 ) {}
 
   @Post('register')
-  async register(@Body() userDto: RegisterDto, @Res({ passthrough: true }) response){
+  async register( @Body() userDto: RegisterDto, @SystemInfo() systemInfo, @Res({ passthrough: true }) response){
```

Comment:
`userDto: UserDto`
`registerDto: RegisterDto`
make sure that variable has the same name as in class, it's more redable